pkg/platform/docker/util.go

```diff
@@ -51,3 +71,254 @@ func ParseDockerCmd(cmd string, kind utils.CmdType, idc Client) (string, string,
 
 	return containerName, networkName, nil
 }
+
+func GenerateDockerEnvs(config ConfigStruct) string {
+	var envs []string
+	for key, value := range config.Envs {
+		envs = append(envs, fmt.Sprintf("-e %s='%s'", key, value))
+	}
+	return strings.Join(envs, " ")
+}
+
+// StartInDocker will check if the docker command is provided as an input
+// then start the Keploy as a docker container and run the command
+// should also return a boolean if the execution is moved to docker
+func StartInDocker(ctx context.Context, logger *zap.Logger, conf *config.Config) error {
+
+	if DockerConfig.Envs == nil {
+		DockerConfig.Envs = map[string]string{
+			"INSTALLATION_ID": conf.InstallationID,
+		}
+	} else {
+		DockerConfig.Envs["INSTALLATION_ID"] = conf.InstallationID
+	}
+
+	err := RunInDocker(ctx, logger)
+	if err != nil {
+		utils.LogError(logger, err, "failed to run the command in docker")
+		return err
+	}
+	// gracefully exit the current process
+	logger.Info("exiting the current process as the command is moved to docker")
```

Comment:
This log was suitable for old implementation, please change this log for new implementation.